src/frontend/js/containers/Login.js

```diff
@@ -0,0 +1,73 @@
+import Button from 'react-bootstrap/lib/Button';
+import FormControl from 'react-bootstrap/lib/FormControl';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import { login } from '../actions/Security';
+
+class Login extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { name: '', password: '' };
+  }
+
+  onChange(e, field) {
+    this.setState({ [field]: e.target.value });
+  }
+
+  onSubmit(e) {
+    e.preventDefault();
+    this.props.onSubmit(this.state);
+  }
+
+  render() {
+    const canSubmit = this.state.name.length > 3 && this.state.password.length > 2;
+    return <form className="login-form" onSubmit={this.onSubmit.bind(this)}>
+      <FormControl
+        disabled={this.props.loggingIn}
+        type="text"
+        onChange={e => this.onChange(e, 'name')}
```

Comment:
This feels a little too generic? I get how this works, but maybe a `onFamilyNameChange` method that delegates down to `this.onChange` would be slightly more readable even if it's a bit more logic? Same for the password `FormControl`...

Reply:
I just try to DRY this a bit because all of them will look exactly the same. I understand the readability argument, but not sure it buys much. And after you understand one of them, all will be the same anyway.